shopify_python/google_styleguide.py

```diff
@@ -0,0 +1,158 @@
+        'C2607': ('Try body too long',
+                  'try-too-long',
+                  "The larger the try body, the more likely that an unexpected exception will be raised"),
+        'C2608': ('Except body too long',
+                  'except-too-long',
+                  "The larger the except body, the more likely that an exception will be raised"),
+        'C2609': ('Finally body too long',
+                  'finally-too-long',
+                  "The larger the except body, the more likely that an exception will be raised"),
+    }
+
+    max_try_exc_finally_body_size = 5
+
+    def visit_assign(self, node):  # type: (astroid.Assign) -> None
+        self.__avoid_global_variables(node)
+
+    def visit_excepthandler(self, node):  # type: (astroid.ExceptHandler) -> None
+        self.__dont_catch_standard_error(node)
+
+    def visit_tryexcept(self, node):  # type: (astroid.TryExcept) -> None
+        self.__minimize_code_in_try_except(node)
+
+    def visit_tryfinally(self, node):  # type: (astroid.TryFinally) -> None
+        self.__minimize_code_in_finally(node)
+
+    def visit_importfrom(self, node):  # type: (astroid.ImportFrom) -> None
+        self.__import_modules_only(node)
+        self.__import_full_path_only(node)
+
+    def visit_raise(self, node):  # type: (astroid.Raise) -> None
+        self.__dont_use_archaic_raise_syntax(node)
+
+    def __import_modules_only(self, node):  # type: (astroid.ImportFrom) -> None
+        """Use imports for packages and modules only."""
+
+        if hasattr(self.linter, 'config') and 'import-modules-only' in self.linter.config.disable:
+            return  # Skip if disable to avoid side-effects from importing modules
+
+        def can_import(module_name):
+            if sys.version_info >= (3, 4):
+                try:
+                    return bool(importlib.util.find_spec(module_name))
+                except AttributeError:
+                    return False
+            else:
+                try:
+                    importlib.import_module(module_name)
+                    return True
+                except ImportError:
+                    return False
+
+        if not node.level and node.modname != '__future__':
+            for name in node.names:
+                name, _ = name
+                parent_module = node.modname
+                child_module = '.'.join((node.modname, name))  # Rearrange "from x import y" as "import x.y"
+                if can_import(parent_module) and not can_import(child_module):
+                    self.add_message('import-modules-only', node=node)
+
+    def __import_full_path_only(self, node):  # type: (astroid.ImportFrom) -> None
+        """Import each module using the full pathname location of the module."""
+        if node.level:
+            self.add_message('import-full-path', node=node)
+
+    def __avoid_global_variables(self, node):  # type: (astroid.Assign) -> None
+        """Avoid global variables."""
```

Comment:
I added a comment below; I assume that implementation details like this should be a comment whereas the purpose of the method should be a docstring.